Unstable test `TestAnalyzeColumnsWithVirtualColumnIndex`

## Enhancement

```
    testing.go:1399: race detected during execution of test
--- FAIL: TestAnalyzeColumnsWithVirtualColumnIndex (55.51s)

WARNING: DATA RACE
Read at 0x00000e066198 by goroutine 23126:
  github.com/pingcap/tidb/pkg/resourcegroup/runaway.(*Manager).RunawayRecordFlushLoop.func1()
      pkg/resourcegroup/runaway/manager.go:162 +0x5b
  github.com/pingcap/tidb/pkg/resourcegroup/runaway.(*Manager).RunawayRecordFlushLoop()
      pkg/resourcegroup/runaway/manager.go:178 +0x5f6
  github.com/pingcap/tidb/pkg/resourcegroup/runaway.(*Manager).RunawayRecordFlushLoop-fm()
      <autogenerated>:1 +0x33
  github.com/pingcap/tidb/pkg/util.(*WaitGroupEnhancedWrapper).Run.func1()
      pkg/util/wait_group_wrapper.go:99 +0xe3

Previous write at 0x00000e066198 by goroutine 70860:
  github.com/pingcap/tidb/pkg/resourcegroup/runaway.(*Manager).RunawayRecordFlushLoop()
      pkg/resourcegroup/runaway/manager.go:159 +0x2f8
  github.com/pingcap/tidb/pkg/resourcegroup/runaway.(*Manager).RunawayRecordFlushLoop-fm()
      <autogenerated>:1 +0x33
  github.com/pingcap/tidb/pkg/util.(*WaitGroupEnhancedWrapper).Run.func1()
      pkg/util/wait_group_wrapper.go:99 +0xe3

Goroutine 23126 (running) created at:
  github.com/pingcap/tidb/pkg/util.(*WaitGroupEnhancedWrapper).Run()
      pkg/util/wait_group_wrapper.go:94 +0x164
  github.com/pingcap/tidb/pkg/domain.(*Domain).runawayStartLoop()
      pkg/domain/runaway.go:78 +0x4ec
  github.com/pingcap/tidb/pkg/domain.(*Domain).runawayStartLoop-fm()
      <autogenerated>:1 +0x33
  github.com/pingcap/tidb/pkg/util.(*WaitGroupEnhancedWrapper).Run.func1()
      pkg/util/wait_group_wrapper.go:99 +0xe3

Goroutine 70860 (running) created at:
  github.com/pingcap/tidb/pkg/util.(*WaitGroupEnhancedWrapper).Run()
      pkg/util/wait_group_wrapper.go:94 +0x164
  github.com/pingcap/tidb/pkg/domain.(*Domain).runawayStartLoop()
      pkg/domain/runaway.go:78 +0x4ec
  github.com/pingcap/tidb/pkg/domain.(*Domain).runawayStartLoop-fm()
      <autogenerated>:1 +0x33
  github.com/pingcap/tidb/pkg/util.(*WaitGroupEnhancedWrapper).Run.func1()
      pkg/util/wait_group_wrapper.go:99 +0xe3
```

--- a/pkg/resourcegroup/runaway/manager.go
+++ b/pkg/resourcegroup/runaway/manager.go
@@ -155,8 +155,8 @@ func (rm *Manager) RunawayRecordFlushLoop() {
 	quarantineRecordCh := rm.quarantineRecordChan()
 	staleQuarantineRecordCh := rm.staleQuarantineRecordChan()
 	flushThreshold := flushThreshold()
-	// recordMap is used to deduplicate records.
-	recordMap = make(map[recordKey]*Record, flushThreshold)
+	// recordMap is used to deduplicate records which will be inserted into `mysql.tidb_runaway_queries`.
+	recordMap := make(map[recordKey]*Record, flushThreshold)
 
 	flushRunawayRecords := func() {
 		if len(recordMap) == 0 {

--- a/pkg/resourcegroup/runaway/record.go
+++ b/pkg/resourcegroup/runaway/record.go
@@ -68,9 +68,6 @@ type Record struct {
 	Repeats int
 }
 
-// recordMap is used to save records which will be inserted into `mysql.tidb_runaway_queries` by function `flushRunawayRecords`.
-var recordMap map[recordKey]*Record
-
 // recordKey represents the composite key for record key in `tidb_runaway_queries`.
 type recordKey struct {
 	ResourceGroupName string

--- a/pkg/resourcegroup/runaway/record_test.go
+++ b/pkg/resourcegroup/runaway/record_test.go
@@ -45,7 +45,7 @@ func TestRecordKey(t *testing.T) {
 	assert.NotEqual(t, hash1, hash3, "Hashes should not be equal for different keys")
 
 	// Test MapKey method
-	recordMap = make(map[recordKey]*Record)
+	recordMap := make(map[recordKey]*Record)
 	record1 := &Record{
 		ResourceGroupName: "group1",
 		SQLDigest:         "digest1",

